### ISSUE TITLE
Can't input more than 9999 MEGA to send

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#4761]
![image](https://github.com/leather-wallet/extension/assets/33360391/46e76276-68da-4cc9-9783-3b1d9c6ed9b7)

I tested some other tokens without proper token metadata, but they didn't have this issue. Also tokens with metadata don't have the same issue either. 

This token does have some of the metadata showing such as token Name/ticker but not the logo. I think this may be caused by some issue with the metadata: checking with Rafael here: https://github.com/hirosystems/token-metadata-api/issues/140#issuecomment-1872254404

![image](https://github.com/leather-wallet/extension/assets/33360391/0979d627-5adc-49dd-93a6-2a4fbd11b042)

![image](https://github.com/leather-wallet/extension/assets/33360391/bfd26beb-518b-4895-b192-953ef4a95704)

Looks like the interface is expecting (or defaults to 6 decimals) whereas the token has 2 decimals as displayed correctly in the lower right corner:

![image](https://github.com/leather-wallet/extension/assets/33360391/93c8b009-c2d8-4b4e-8432-5cbaeaaf4677)

**Comment by 314159265359879** [comment]
Rafael has updated the metadata but that didn't fix the issue in Leather, there is no change, I am not sure what is causing this. 

Metadata looks not to different from other tokens
https://api.hiro.so/metadata/v1/ft/SP3D6PV2ACBPEKYJTCMH7HEN02KP87QSP8KTEH335.mega

Most other tokens have 6 or 8 decimals, mega has 2. 
cached image looks accessable: https://ipfs.io/ipfs/Qmdgks1HjYZQhF4sTnkoeh7naic7J3G5aHQk91Uq25RwmF
![Qmdgks1HjYZQhF4sTnkoeh7naic7J3G5aHQk91Uq25RwmF](https://github.com/leather-wallet/extension/assets/33360391/0851d861-7923-4dda-93b6-9e97b7f194e3)

**Comment by markmhendrickson** [comment]
@alter-eggo is this issue also resolved by https://github.com/leather-wallet/extension/pull/5048?

**Comment by alter-eggo** [comment]
@markmhendrickson yes, user will be able to input more, but I haven't  touched the problem with icons, that @314159265359879 mentions here
